core/src/main/java/org/mskcc/cbio/portal/dao/DaoGeneOptimized.java

```diff
@@ -338,7 +339,16 @@ public CanonicalGene getNonAmbiguousGene(String geneId) {
      * @param chr chromosome
      * @return a gene that can be non-ambiguously determined, or null if cannot.
      */
-    public CanonicalGene getNonAmbiguousGene(String geneId, String chr) {
+    public CanonicalGene getNonAmbiguousGene(String geneId, String chr) { return getNonAmbiguousGene(geneId, chr, null); }
+
+        /**
+         * Look for gene that can be non-ambiguously determined.
+         * @param geneId an Entrez Gene ID or HUGO symbol or gene alias
+         * @param chr chromosome
+         * @param pMonitor for import warnings
+         * @return a gene that can be non-ambiguously determined, or null if cannot.
+         */
+    public CanonicalGene getNonAmbiguousGene(String geneId, String chr, ProgressMonitor pMonitor) {
```

Comment:
I don't like to put pMoniter in the function signature. Maybe we should have a ProgressMonitor singleton class that script and dao classes can call? And the singleton class is only initialized when importing (not the web code).